affinity/cpp-20/d0796r1.md

```diff
@@ -165,94 +206,163 @@ There are some additional challenges which we have been investigating but are no
 
 ## Class `execution_resource`
 
-The `execution_resource` class provides an abstraction over a system's hardware capable of memory allocation, execution of light weight exeution agents or both.
+The `execution_resource` class provides an abstraction over a system's hardware capable of memory allocation, execution of light weight exeution agents or both. An `execution_resource` can represent further `execution_resource`s, these `execution_resource`s are said to be *members of* this `execution_resource`.
 
-### `execution_resource` constructors
+> [*Note:* The `execution_resource` is required to be implemented such that the underlying software abstraction is initialised on when the `execution_resource` is constructed, maintained through reference counting and cleaned up on destruction of the final reference. *--end note*]
 
-	execution_resource() = delete;
+### `execution_resource` constructors
 
+    execution_resource();
 
-[*Note:* An implementation of `execution_resource` is permitted to provide non-public constructors to allow other objects to construct them. *--end note*]
+> [*Note:* An implementation of `execution_resource` is permitted to provide non-public constructors to allow other objects to construct them. *--end note*]
 
 ### `execution_resource` assignment
 
-The `execution_resource` class is not `CopyConstructible` (C++Std [copyconstructible]).
-
-    execution_resource(const execution_resource &) = delete;
-    execution_resource(execution_resource &&) = delete;
-    execution_resource &operator=(const execution_resource &) = delete;
-    execution_resource &operator=(execution_resource &&) = delete;
+    execution_resource(const execution_resource &);
+    execution_resource(execution_resource &&);
+    execution_resource &operator=(const execution_resource &);
+    execution_resource &operator=(execution_resource &&);
 
 ### `execution_resource` destructor
 
-The `execution_resource` class is not `Destructible` (C++Std [destructible]).
-
-	~execution_resource() = delete;
+    ~execution_resource();
 
 ### `execution_resource` operations
 
-      size_t concurrency() const noexcept;
+    size_t concurrency() const noexcept;
 
-*Returns:*
+*Returns:* The collective concurrency available to this resource. More pecifically, the number of *threads of execution* collectively available to this `execution_resource` and any resources which are *members of*, recursively.
 
-      size_t partition_size() const noexcept;
+    std::vector<resource> resources() const noexcept;
 
-*Returns:*
+*Returns:* All `execution_resource`s which are *members of* this resource.
 
-      const execution_resource &partition(size_t i) const noexcept;
+    const execution_resource &member_of() const noexcept;
 
-*Returns:*
+*Returns:* The `execution_resource` which this resource is a *member of*.
 
-      const execution_resource &member_of() const noexcept;
+    std::string name() const noexcept;
```

Comment:
can this return a string and be noexcept at the same time? what is the expected value of std::string if the system cannot allocate memory for the string for example? are you expecting a terminate?

Reply:
That's an interesting point, though I would expect that an implementation would have a string allocated on construction so that `name` would not have to perform any allocation.